rsmtool/utils/commandline.py

```diff
@@ -0,0 +1,321 @@
+    # always need a configuration file
+    parser_run.add_argument('config_file',
+                            type=existing_configuration_file,
+                            help=f"the {name} JSON configuration file to run")
+
+    # if it uses an output directory, let's add that
+    if uses_output_directory:
+        parser_run.add_argument('output_dir',
+                                nargs='?',
+                                default=os.getcwd(),
+                                help="the output directory where all the files "
+                                     "for this run will be stored")
+
+    # if it allows overwrting the output directory, let's add that
+    if allows_overwriting:
+        parser_run.add_argument('-f',
+                                '--force',
+                                dest='force_write',
+                                action='store_true',
+                                default=False,
+                                help=f"if specified, {name} will overwrite the "
+                                     f"contents of the output file or directory "
+                                     f"even if it contains the output of a "
+                                     f"previous run ")
+
+    # add any extra options passed in for the rub subcommand;
+    for parser_option in extra_run_options:
+
+        # construct the arguments and keyword arguments needed for the
+        # `add_argument()` call to the parser
+        argparse_option_args = []
+        argparse_option_kwargs = {}
+
+        # first add the destination and the help string
+        argparse_option_kwargs["dest"] = f"{parser_option.dest}"
+        argparse_option_kwargs["help"] = f"{parser_option.help}"
+
+        # now add any optional information
+        if parser_option.shortname is not None:
+            argparse_option_args.append(f"-{parser_option.shortname}")
+        if parser_option.longname is not None:
+            argparse_option_args.append(f"--{parser_option.longname}")
+        if parser_option.action is not None:
+            argparse_option_kwargs['action'] = f"{parser_option.action}"
+        if parser_option.default is not None:
+            argparse_option_kwargs["default"] = f"{parser_option.default}"
+        if parser_option.required is not None:
+            try:
+                assert type(parser_option.required) == bool
+            except AssertionError:
+                raise TypeError(f"the 'required' field for CmdOption must be "
+                                f"boolean, you specified '{parser_option.required}'")
+            else:
+                argparse_option_kwargs["required"] = parser_option.required
+        if parser_option.nargs is not None:
+            argparse_option_kwargs['nargs'] = f"{parser_option.nargs}"
+
+        # add this argument to the parser
+        parser_run.add_argument(*argparse_option_args, **argparse_option_kwargs)
+
+    return parser
+
+
+def generate_configuration(name,
```

Comment:
Should this be 'context' for consistency with the rest of the code?

Reply:
Done.